### PR TITLE
Avoid unncessary signed/unsigned conversion

### PR DESCRIPTION
**Pull request by tautschnig**

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 0a063b37d453bf52353df046c1df2256cc3ec68a).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78358180

File: src/goto-instrument/accelerate/polynomial.cpp [hunk 1]
```diff
@@ -138,15 +138,11 @@ void polynomialt::from_expr(const exprt &expr)
...
   else if(expr.id()==ID_constant)
   {
-    mp_integer mp;
-    unsigned int l;
-    constant_exprt const_expr=to_constant_expr(expr);
-
-    mp=binary2integer(const_expr.get_value().c_str(), true);
-    l=mp.to_long();
+    const constant_exprt &const_expr = to_constant_expr(expr);
+    const mp_integer mp = binary2integer(const_expr.get_value().c_str(), true);
```
Comment (kroening) [inline src/goto-instrument/accelerate/polynomial.cpp]: This should really be numeric_cast<int>(const_expr)

Comment (tautschnig) [reply src/goto-instrument/accelerate/polynomial.cpp]: Done.

**Review by kroening — APPROVED** [review]

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 6dc6ea99cdc72ec3756da03b978fc254229f3061).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/78387196

**Review by peterschrammel — APPROVED** [review]

**Pull request merged**